### ISSUE TITLE
MSAL.js login reuses previous user token

### ISSUE THREAD

**Issue by MarinaGurevich** [AzureAD/microsoft-authentication-library-for-js#2834]
- We have a hybrid application that is first asking users to enter their email address, then based on whether their email domain supports the Federated Authentication feature we direct them to Azure AD B2C login. If it doesn't we ask them for their internal application username and password stored in the database. We are not using B2C "local accounts" feature, only external IDPs. Currently we only support Azure AD as external IDP, but we are planning to introduce others. Since we only support one IDP per customer we are using direct-sign-in and passing user's email as login_hint and their email domain as domain_hint in MSAL.js call. Domain_hint is then mapped in custom policies to their Azure AD tenant sign-in user flow. So users are not presented by the B2C screen, but are taken directly to the external IDP login.
- We want to support SSO and single sign out is out of question, so if user signs out of our app they are not signed out of their respective Azure AD completely and can still use it with other applications.
- The issue is that if a user browser is shared by multiple employees, which is a very likely scenario with our customers, if one user logs out of their application and then 2nd user tries to login to the same Azure AD domain, but with a different email, although I can see their tokens deleted from local storage, they are still able to sign-in with previous user token, without even being presented with a login prompt, even though we pass a different login_hint.
- Adding prompt: 'login' doesn't help and makes no difference at all, possibly because we are using external IDP.
- In the Chrome browser tools Console after "aquiring token interactive" I see the message "user session already exists, login not required", although login_hint in the MSAL request has changed. 
- I understand that session is managed cloud-side in the AAD, and if there is an exiting AAD session it reuses it, but shouldn't it recognize that login_hint has changed?

const loginRequest = {
  scopes: ["openid profile"],
  loginHint: 'marina.gurevich@heliosb2c.com', - changes dynamically based on user email
  extraQueryParameters: { domain_hint: 'heliosb2c_com' } - changes dynamically based on email domain
};

Any ideas?

**Comment by jo-arroyo** [comment]
@MarinaGurevich Is there an amount of time that affects this behavior? Could you also provide the version of MSAL you are using? Using our [issue templates](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/new/choose) can help us diagnose where the problem is.

@nickgmicrosoft Can you take a look at this? Is it related to the B2C service?

**Comment by MarinaGurevich** [comment]
I am reproducing the issue with the sample pack https://github.com/Azure-Samples/active-directory-b2c-javascript-msal-singlepageapp I modified for two subsequent users signing in. This is MSAL.js 1.0 since 2.0 was not supported by B2C till now.

I did not measure the time. It is definitely related to the Azure AD session still going, but for our corporate Azure AD where multiple applications are connected to it I think there always be a session going.

Personally I just found a workaround to pass prompt: 'login' in MSAL.js request if user email changes, but for that to work with B2C external IDP such as Azure AD I had to add <InputClaim ClaimTypeReferenceId="prompt" DefaultValue="{OIDC:prompt}" AlwaysUseDefaultValue="true" /> to the AAD Claims Provider.

But other people might still have this issue.
